lib/stack.rb

```diff
@@ -1,19 +1,20 @@
+require_relative './linked_list.rb'
+
 class Stack
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = LinkedList.new
   end
 
   def push(element)
-    raise NotImplementedError, "Not yet implemented"
+    return @store.add_last(element)
   end
 
   def pop
-    raise NotImplementedError, "Not yet implemented"
+    return @store.remove_last()
   end
 
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    return true if @store.get_last.nil?
```

Comment:
This works, but `LinkedList` also has a `.empty?` method.

```suggestion
    return @store.empty?
```